src/framework/FormatChecker.ts

```diff
@@ -37,11 +37,8 @@ export class FormatChecker extends Base {
 
   async _messageHandler(message: Message): Promise<void> {
     if (this.bot === undefined) return;
-    const client = this.bot.client;
-    if (
-      message.channel.id !== findTextChannelByName(client, this.channelName)?.id
-    )
-      return;
+    if (!(message.channel instanceof TextChannel)) return;
+    if (message.channel.name !== this.channelName) return;
```

Comment:
C'est déjà ce que fait findTestChannelByName

Reply:
Je trouve que c'est inutile de chercher le salon alors qu'il suffit seulement de vérifier le nom.

Reply:
On compare pas juste les noms, mais les IDs. A voir ce qu'en pense @targos ?

Reply:
Ça sert à rien de comparer les ID ici, puisqu'en gros on compare un nom et un ID. La comparaison est donc inutile ici.
Donc tout ce que rapportait le `findTextChannelByName` c'est la vérification du type du channel (qui a maintenant été remplacé)

Reply:
Par souci de rigueur, je ne comparerais pas simplement les noms (pas uniques).

Reply:
Il n'y a aucune différence avec ce que tu fais.
Si tu essaies de comparer un nom (`this.channelName` + `findTextChannelByName`) et une ID de cette façon tu perds le concept d'unicité. Et encore pire, cela peut poser des problèmes, il peut y avoir plusieurs salons nommés `liens` (par exemple) dans un même serveur, le `findTextChannelByName` va renvoyer le premier salon qu'il croise se nommant ainsi, les deux salons (tous les deux nommés `liens`) n'ont pas la même ID, ce qui fait échouer le test.

Reply:
Alors ajoute un helper, c'est du code potentiellement redondant.

Reply:
Un helper pour ? Pour vérifier si un canal est de type TextChannel ou si un canal est de type TextChannel et a un nom x ?